app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -7,108 +7,95 @@ import { Section } from 'components/ucsf/Section/Section';
-        ...targetEligibility,
-        checked: !eligibilityGroup.eligibilities[eligibilityIndex].checked,
+      const updatedEligibilities = {
+        ...selectedEligibilities,
+        [eligibilityCheckedId]: targetValue,
       };
 
-      updatedEligibilities = [
-        { ...seeAllEligibility },
-        ...eligibilityGroup.eligibilities.slice(1, eligibilityIndex),
-        updatedEligibility,
-        ...eligibilityGroup.eligibilities.slice(eligibilityIndex + 1),
-      ];
-    }
+      // If target checked value is false, uncheck "See All" box as well
+      if (!targetValue) {
+        // N.B. Use of "!" assumes that every Eligibility array will have a "See All" element
+        updatedEligibilities[seeAllEligibilityItem!.checkedId] = false;
+      }
 
-    return {
-      ...eligibilityGroup,
-      eligibilities: updatedEligibilities,
-    };
+      setSelectedEligibilities(updatedEligibilities);
+    }
   };
 
   // Toggles all eligibilities in accordance with the toggleState argument
   const massToggleGroupEligibilities = (
     eligibilities: Eligibility[],
     toggleState: boolean,
-  ) => (
-    eligibilities.map((eligibility: Eligibility) => ({
-      ...eligibility,
-      checked: toggleState,
-    }))
-  );
+  ) => {
+    const updatedEligibilities = {
+      ...selectedEligibilities,
+    };
+
+    eligibilities.forEach(eligibility => {
+      updatedEligibilities[eligibility.checkedId] = toggleState;
+    });
+
+    setSelectedEligibilities(updatedEligibilities);
+  };
 
   return (
     <div className={styles.eligibilitiesBox}>
       <div className={styles.eligibilitiesBox_title}>Client Identity</div>
       <ol className={styles.eligibilitiesLabels}>
-
-        {/* Todo: This list rendering logic will be refactored when the API is setup */}
-        {eligibilityGroupList.map((eligibilityGroup, index) => (
+        {eligibilityGroupList.map(eligibilityGroup => (
           <li key={eligibilityGroup.label} className={styles.listContainer}>
             <span className={styles.eligibilityGroupLabel}>
               {eligibilityGroup.label}
             </span>
             <ul className={styles.eligibilitiesList}>
-              {eligibilityGroup.eligibilities.map((eligibility, i) => (
+              {eligibilityGroup.eligibilities.map(eligibility => (
                 <li key={`${eligibilityGroup.label}-${eligibility.name}`} className={styles.eligibilityGroup}>
                   <Checkbox
-                    onChange={() => setEligibilityGroup(index, toggleChecked(eligibilityGroup, i))}
+                    onChange={() => handleEligibilityClick(
+                      eligibility,
+                      eligibilityGroup.eligibilities,
+                    )}
                     name={eligibilityGroup.label}
                     id={`${eligibilityGroup.label}-${eligibility.name}`}
-                    checked={eligibility.checked}
+                    checked={selectedEligibilities[eligibility.checkedId] || false}
```

Comment:
```suggestion
                    checked={selectedEligibilities[eligibility.checkedId] ?? false}
```

I forget if we had issues with tooling, but it'd be better to use `??` over `||` here, since it more explicitly indicates that you just want a fallback for the case where the key doesn't exist, in which case an `undefined` will be returned.